com.unity.ml-agents.extensions/Runtime/Sensors/ArticulationBodyJointExtractor.cs

```diff
@@ -0,0 +1,144 @@
+#if UNITY_2020_1_OR_NEWER
+
+using System.Collections.Generic;
+using UnityEngine;
+using Unity.MLAgents.Sensors;
+
+namespace Unity.MLAgents.Extensions.Sensors
+{
+    public class ArticulationBodyJointExtractor : IJointExtractor
+    {
+        ArticulationBody m_Body;
+
+        public ArticulationBodyJointExtractor(ArticulationBody body)
+        {
+            m_Body = body;
+        }
+
+        public int NumObservations(PhysicsSensorSettings settings)
+        {
+            return NumObservations(m_Body, settings);
+        }
+
+        public static int NumObservations(ArticulationBody body, PhysicsSensorSettings settings)
+        {
+            if (body == null || body.isRoot)
+            {
+                return 0;
+            }
+
+            var totalCount = 0;
+            if (settings.UseJointPositions)
+            {
+                switch (body.jointType)
+                {
+                    case ArticulationJointType.RevoluteJoint:
```

Comment:
What happens in this case? Is this TODO later?

Reply:
That should "fall through" to `totalCount += 2 * body.dofCount` right? Or is my understanding of switch statements obsolete?

I believe (should verify with the physics folks) that:
RevoluteJoint: 1 angular dof
SphericalJoint: 3 angular dof
FixedJoint: 0 dof
PrismaticJoint: 1 linear dof

so both RevoluteJoint and SphericalJoint are basically the same (just different values for body.dofCount)

Reply:
fall through is what will happen, maybe a comment to display intent could help